src/Analyzers/CSharp/Tests/UseCollectionExpression/UseCollectionExpressionForBuilderTests.cs

```diff
@@ -1589,4 +1589,122 @@ IEnumerable<int> M()
+    [WorkItem("https://github.com/dotnet/roslyn/issues/74208")]
+    public async Task TestLeadingComment(string pattern)
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = $$"""
+                using System.Collections.Immutable;
+
+                class C
+                {
+                    void M()
+                    {
+                        //Leading
+                        {{pattern}}
+                        [|builder.Add(|]0);
+                        ImmutableArray<int> array = builder.ToImmutable();
+                    }
+                }
+                """ + s_arrayBuilderApi,
+            FixedCode = """
+                using System.Collections.Immutable;
+
+                class C
+                {
+                    void M()
+                    {
+                        //Leading
+                        ImmutableArray<int> array = [0];
+                    }
+                }
+                """ + s_arrayBuilderApi,
+            LanguageVersion = LanguageVersion.CSharp12,
+            ReferenceAssemblies = ReferenceAssemblies.Net.Net80,
+        }.RunAsync();
+    }
+
+    [Theory, MemberData(nameof(SuccessCreationPatterns))]
+    [WorkItem("https://github.com/dotnet/roslyn/issues/74208")]
+    public async Task TestLeadingCommentWithAdditionalTrivia1(string pattern)
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = $$"""
+                using System.Collections.Immutable;
+
+                class C
+                {
+                    void M()
+                    {
+                        //Leading
+                        {{pattern}}
+                        [|builder.Add(|]0);
+                        //comment
+                        ImmutableArray<int> array = builder.ToImmutable();
+                    }
+                }
+                """ + s_arrayBuilderApi,
+            FixedCode = """
+                using System.Collections.Immutable;
+
+                class C
+                {
+                    void M()
+                    {
+                        //Leading
+                        //comment
+                        ImmutableArray<int> array = [0];
+                    }
+                }
+                """ + s_arrayBuilderApi,
+            LanguageVersion = LanguageVersion.CSharp12,
+            ReferenceAssemblies = ReferenceAssemblies.Net.Net80,
+        }.RunAsync();
+    }
+
+    [Theory, MemberData(nameof(SuccessCreationPatterns))]
+    [WorkItem("https://github.com/dotnet/roslyn/issues/74208")]
+    public async Task TestLeadingCommentWithAdditionalTrivia2(string pattern)
+    {
+        await new VerifyCS.Test
+        {
+            TestCode = $$"""
+                using System.Collections.Immutable;
+
+                class C
+                {
+                    void M()
+                    {
+                        //Leading
+                        {{pattern}}
+                        //Leading
```

Comment:
Name leading trivias differently for clarity, e.g. `// Leading 1` and `// Leading 2`

Reply:
Fixed!